### PR TITLE
Make end-to-end tests work with non-zero fees, and with Devnet.

### LINKED ISSUE #1620: Make end-to-end tests work with Devnet.
The end-to-end tests (except those that require control of the admin chain) should be general enough to run with a local network as well as the Devnet. That means they need to support different fee structures, optionally use the faucet, and not rely on using root chains.

### PR DESCRIPTION
**Pull request by afck**
## Motivation

The end-to-end tests should work both with a local network and with the Devnet.

## Proposal

Make the local tests use fees, too, and fix the tests accordingly.

## Test Plan

I ran it against Devnet. In CI, it has fees now, too.

## Links

- Closes #1620.
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-service/src/cli_wrappers/local_net.rs [hunk 1]
```diff
@@ -124,23 +126,14 @@
             database,
             network,
             num_other_initial_chains: 10,
-            initial_amount: Amount::from_tokens(10),
+            initial_amount: Amount::from_tokens(1_000_000),
+            policy: ResourceControlPolicy::devnet(),
```
Comment (andresilva91) [inline linera-service/src/cli_wrappers/local_net.rs:130]: I'm interested in what could cause this lint error once you find it out 😅🤔

Comment (afck) [reply linera-service/src/cli_wrappers/local_net.rs:130]: `linera-service`'s `test` feature didn't depend on `linera-execution/test`. I added a fix.

File: linera-base/src/data_types.rs [hunk 2]
```diff
@@ -466,6 +493,16 @@ impl Amount {
         Amount(10u128.pow(Amount::DECIMAL_PLACES as u32 - 3)).saturating_mul(millitokens)
     }
 
+    /// Returns an `Amount` corresponding to that many microtokens, or `Amount::MAX` if saturated.
+    pub fn from_micro(microtokens: u128) -> Amount {
```
Comment (ma2bd) [inline linera-base/src/data_types.rs:497]: Should these names be plural? `from_{milli,micro,nano,atto}s` or even `from_{milli,micro,nano,atto}tokens` ?

Comment (ma2bd) [reply linera-base/src/data_types.rs:497]: Perhaps native speakers can chime in @christos-h @Twey

File: linera-execution/src/policy.rs [hunk 3]
```diff
@@ -146,4 +146,20 @@ impl ResourceControlPolicy {
             ..Self::default()
         }
     }
+
+    /// Creates a policy that matches the Devnet launched in January 2024.
```
Comment (ma2bd) [inline linera-execution/src/policy.rs:150]: I think we'd like this to match a (the most) recent devnet.

**Review by ma2bd — APPROVED** [review]
Thanks @afck for taking care of this!

File: linera-execution/src/policy.rs (hunk 3 above)
Comment (afck) [reply linera-execution/src/policy.rs:150]: It still does, until we use https://github.com/linera-io/linera-infra/pull/73.

But I guess it doesn't hurt to already add the updated fees? I'll add a commit.

File: linera-base/src/data_types.rs (hunk 2 above)
Comment (afck) [reply linera-base/src/data_types.rs:497]: I agree, that's also what the standard library does. Standard library trumps native speakers!

File: linera-execution/src/policy.rs (hunk 3 above)
Comment (afck) [reply linera-execution/src/policy.rs]: I had to add the approximate balance assertions. I think they do make sense in tests where we want to check whether a transfer of an integer token amount has arrived.

Comment (afck) [reply linera-execution/src/policy.rs]: But what's the alternative? E.g. in the faucet test, shouldn't we test that the faucet put the expected balance into the new chains?

Comment (ma2bd) [reply linera-execution/src/policy.rs]: In this case, why is it approximate?

Comment (afck) [reply linera-execution/src/policy.rs]: You're right, those should be exact. For the others we can maybe use the owner account: https://github.com/linera-io/linera-protocol/issues/1637
I'll work on this tomorrow.

**Pull request merged**